src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,244 @@
+                throw new TobiasException("Kindly type your task a space after deadline!");
+            }
+
+            int byIndex = command.indexOf("/by");
+
+            if (byIndex == -1) {
+                throw new TobiasException("Hey, please use '/by' to specify the deadline!!");
+            }
+
+            String description = command.substring(9, byIndex);
+
+            int deadlineIndex = byIndex+4;
+
+            if (deadlineIndex > command.length()) {
+                throw new TobiasException("Hey, please enter a date & time in this format : dd-MM-yyyy HHmm !");
+            }
+
+            String deadline = command.substring(byIndex + 4);
+
+            LocalDateTime dd = dateFromString(deadline);
+
+            if (deadline.isEmpty() || deadline.isBlank()) {
+                throw new TobiasException("Hey, please enter a date & time in this format : dd-MM-yyyy HHmm !");
+            }
+
+            return new DeadlineCommand(description, dd);
+        }
+        else if (command.startsWith("event")) {
+            if (command.equals("event")) {
+                throw new TobiasException("Hey, please enter a description !");
+            }
+
+            String blank = command.substring(5,6);
+
+            if (!blank.isBlank()) {
+                throw new TobiasException("Kindly type your task a space after event!");
+            }
+
+            int fromIndex = command.indexOf("/from");
+            int toIndex = command.indexOf("/to");
+
+            if (fromIndex == -1 && toIndex == -1) {
+                throw new TobiasException("Hey, please enter a /from and /to date/day/time!!");
+            }
+
+            if (fromIndex == -1) {
+                throw new TobiasException("Hey, please enter a /from date/day/time!!");
+            }
+
+            if (toIndex == -1) {
+                throw new TobiasException("Hey, please enter a /to date/day/time!!");
+            }
+
+            if (fromIndex > toIndex && toIndex > 0) {
+                throw new TobiasException("Hey, you have to enter /from before /to !!");
+            }
+
+            String from = command.substring(fromIndex + 6, toIndex);
+
+            if (from.isEmpty() || from.isBlank()) {
+                throw new TobiasException("Hey, please enter a /from date/day/time!!");
+            }
+
+            if ((toIndex + 4) > command.length()) {
+                throw new TobiasException("Hey, please enter a /to date/day/time!!");
+            }
+
+            String to = command.substring(toIndex + 4);
+
+            if (to.isEmpty() || to.isBlank()) {
+                throw new TobiasException("Hey, please enter a /to date/day/time!!");
+            }
+
+            String description = command.substring(6, fromIndex);
+
+            LocalDateTime f = dateFromString(from);
+            LocalDateTime t = dateFromString(to);
+
+            return new EventCommand(description, f, t);
+        }
+        else if (command.startsWith("mark")) {
```

Comment:
Referring to the [Coding Standard](https://se-education.org/guides/conventions/java/intermediate.html) under 

>  The if-else class of statements should have the following form:

```suggestion
        } else if (command.startsWith("mark")) {
```

Reply:
There are some similar structures below as well.